swap_meet/vendor.py

```diff
@@ -1,2 +1,64 @@
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory = None):
+        if not inventory:
+            inventory = []
+        self.inventory = inventory
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self, item):
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+
+    def get_by_category(self, category):
+        """gets items in inventory of given category"""
+        items_by_category = []
+        
+        for item in self.inventory:
+            if item.category == category:
+                items_by_category.append(item)
+
+        return items_by_category
+    
+    def swap_items(self, other, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other.inventory:
+            return False
+
+        my_index = self.inventory.index(my_item)
+        their_index = other.inventory.index(their_item)
+
+        self.inventory[my_index], other.inventory[their_index] = other.inventory[their_index], self.inventory[my_index]
+        
+        return True
+    
+    def swap_first_item(self, other):
+        if not self.inventory or not other.inventory:
+            return False
+
+        my_item = self.inventory[0]
+        their_item = other.inventory[0]
+
+        return self.swap_items(other, my_item, their_item)
```

Comment:
👍 

Nice reuse of your swap method. Notice that because of your index and swap approach in `swap_items`, we can basically guarantee that `swap_first_items` is O(1) time complexity, since the called function will always find these items in the first location in the inventory lists, and the swap is also constant time. An approach using `remove` and `append` would still be O(n) due to the shifting that a remove would require, even if the item is in the first position.